BuildForSDG/team-029-backend
# sikiru-ft-admin-roles

## Description
Implementation of some of the admin roles as described in the following issues:
Fixes #36 
Fixes #34 
Fixes #33 
Fixes #30 
Fixes #27 

## How Has This Been Tested?
- Checkout this branch
- Run `db-migrate up`
- Make a POST request to { base_url } /api/v1/road/create as an ADMIN with sample body {
    title: ikeja
}
- Expect a road to be created and returned
- Make a PATCH request to { base_url } /api/v1/road/update-title  as an ADMIN with sample body {
    title: ikeja along
}
- Expect the road to be updated with the new title and returned

- Make a GET request to { base_url } /api/v1/road/all as an ADMIN 
- Expect all the roads within current pagination limit to fetch the next page us /api/v1/road/all?page=2 and so on

- Make a GET request to { base_url }/api/v1/road/detail/2 as an ADMIN or ROAD_WARDEN 
- Expect the details of the road along with the assigned wardens to be returned (all road wardens ever assigned or removed will be returned)

- Make a POST request to { base_url } /api/v1/road/remove as an ADMIN with sample body {
   assignment_id:5
   road_id:1
}
- Expect the warden to be removed from the assignment thereby setting its status to Inactive and date_unassigned to the current date

- Make a POST request to { base_url} /api/v1/road/assign as an ADMIN with sample body {
   road_id:1
   warden_id:USR-44C87
}
- Expect the warden to be assigned to the road successfully

- Make a GET request to { base_url } /api/v1/warden/information/USR-C4855 as an ADMIN or ROAD_WARDEN 
- Expect the warden's information to be returned along with his road history

## Checklist:
